lib/src/routing/connection_registry.rs

```diff
@@ -1,350 +1,221 @@
-                                    ttl - (now.elapsed().as_secs() % table.ttl)
-                                }
-                                Err(e) => {
-                                    error!("Failed to refresh routing table: {}", e);
-                                    // we don't want to lose the initial TTL synchronization: if the forced update is triggered,
-                                    // we derive the TTL from the initial time. Example:
-                                    // if the TTL is 60 seconds and the forced update is triggered after 10 seconds,
-                                    // we want to set the TTL to 50 seconds, so that the next update will be in 50 seconds
-                                    ttl - (now.elapsed().as_secs() % ttl)
-                                }
-                            };
-                        }
-                        Some(RegistryCommand::Stop) | None => {
-                            debug!("Stopping background updater");
-                            break;
-                        }
+            } else {
+                match self
+                    .fetch_routing_table(db.clone(), imp_user.clone(), bookmarks)
+                    .await
+                {
+                    Ok(new_table) => {
+                        let database_table: DatabaseTable = new_table.into();
+                        let servers = database_table.resolve();
+                        debug!("Routing table for database {db_name} refreshed");
+                        servers
+                    }
+                    Err(e) => {
+                        error!("Failed to refresh routing table for database {db_name}: {e}");
+                        vec![]
                     }
                 }
             }
-
-            debug!("Resetting interval with TTL: {}", ttl);
-            // recreate interval with the new TTL or the derived one in case of a forced update
-            interval = tokio::time::interval(Duration::from_secs(ttl));
-            interval.tick().await;
-        }
-    });
-    tx
-}
-
-impl ConnectionRegistry {
-    /// Retrieve the pool for a specific server and database.
-    pub fn get_pool(&self, server: &BoltServer) -> Option<ConnectionPool> {
-        self.pool_registry
-            .get(server)
-            .map(|pool| pool.value().clone())
-    }
-
-    /// Mark a server as available for a specific database.
-    pub fn mark_unavailable(&self, server: &BoltServer, db: Option<Database>) {
-        let db_name = self.get_db_name(db);
-        if self.databases.contains_key(db_name.as_str()) {
-            if let Some(index) = self
-                .databases
-                .get(db_name.as_str())
-                .and_then(|vec| vec.iter().position(|s| server.has_same_address(s)))
+        } else {
+            match self
+                .fetch_routing_table(db.clone(), imp_user.clone(), bookmarks)
```

Comment:
`db` and `imp_user` can be used directly without a clone

```suggestion
                .fetch_routing_table(db, imp_user, bookmarks)
```